marker_api/src/span.rs

```diff
@@ -235,7 +235,7 @@ impl<'ast> std::fmt::Debug for Span<'ast> {
                 fmt_pos(file.try_to_file_pos(self.end))
             ),
             SpanSource::Macro(expn) => format!("[Inside Macro] {:#?}", expn.call_site()),
-            SpanSource::Buildin(_) => "[From Prelude]".to_string(),
+            SpanSource::Builtin(_) => "[From Prelude]".to_string(),
```

Comment:
If lints can't observe the prelude, then we may go without the "Builtin" span source? Or is there any other way this "Builtin" span source can be created? The test that was added in this PR no longer contains such a span source. If there is any other way it could be created, it would be good to have a test that showcases how it can be constructed

Reply:
The filter mechanism added in this PR should filter out all spans with a builtin source. However, I'm not 100% sure if that really removes every trace. Before this new `SpanSource` Marker would ICE when requesting these spans. I'm in favor of keeping it in the API as it will prevent potential ICEs. If we're starting to look at the stabilization of Marker, we can review if this value ever came up again.

I can imagine that also some unstable features might fall back on this source, during initial development. This is just a guess, though.

I'll change the text of this function to say `"[Builtin]"` instead.